File: backend/app/schemas/anexo.py
from datetime import datetime

from pydantic import BaseModel


class AnexoResponse(BaseModel):
    id: int
    chamado_id: int
    usuario_id: int
    nome_original: str
    content_type: str
    tamanho_bytes: int
    criado_em: datetime

    model_config = {"from_attributes": True}

class AnexoDownloadResponse(BaseModel):
    url: str
    expira_em_segundos: int

File: frontend/src/types/anexo.ts
export interface Anexo {
  id: number;
  chamado_id: number;
  usuario_id: number;
  nome_original: string;
  content_type: string;
  tamanho_bytes: number;
  criado_em: string;
}

export interface AnexoDownload {
  url: string;
  expira_em_segundos: number;
}